doc/api/http.md

```diff
@@ -157,6 +157,33 @@ socket/stream from this function, or by passing the socket/stream to `callback`.
 
 `callback` has a signature of `(err, stream)`.
 
+### agent.keepSocketAlive(socket)
+<!-- YAML
+added: REPLACEME
+-->
+
+Called when socket is detached from a request and could be persisted by the
+Agent. Default behavior is to:
+
+* `socket.unref()`
+* `socket.setKeepAlive()` with the `agent.keepAliveMsecs` as an argument
+
+This method can be overridden by particular `Agent` class child. If this method
```

Comment:
s/by/by a/

Also, I'm not sure that "Agent class child" is the best terminology? What about 'subclass' or 'implementation'?